taichi/aot/graph_data.cpp

```diff
@@ -23,7 +23,7 @@ void CompiledGraph::run(
       TI_ERROR_IF(found == args.end(), "Missing runtime value for {}",
                   symbolic_arg.name);
       const aot::IValue &ival = found->second;
```

Comment:
should we first assert the types are the same?

Reply:
good point! added the asserts below since the runtime types may map to different compile time type. (like texture -> rwtexture & texture)